pkg/proxy/types.go

```diff
@@ -42,3 +43,22 @@ type ServicePortName struct {
 func (spn ServicePortName) String() string {
 	return fmt.Sprintf("%s:%s", spn.NamespacedName.String(), spn.Port)
 }
+
+type ServiceInfo interface {
```

Comment:
comments comments comments, please.  Types and each method.

Reply:
Soooory for delay, comments are added now.

Reply:
Please don't be angry about the lack of comments, I was just not sure if you like this refactor...

Reply:
I still hope we can find a better name than "Info" ?  Ideas?  `ServicePort` ?  `ServicePortHandle` ?

Reply:
Yeah, `ServicePort` seems to be better. 

It's fixed now. PTAL.